grpc-dump/dump/dump_interceptor.go

```diff
@@ -42,15 +68,19 @@ func dumpInterceptor(logger logrus.FieldLogger, output io.Writer, decoder proto_
 		}
 
 		var err error
-		for _, message := range rpc.Messages {
-			message.Message, err = decoder.Decode(info.FullMethod, message)
+		for i := range rpc.Messages {
+			msg, err := decoder.Decode(info.FullMethod, rpc.Messages[i])
 			if err != nil {
 				logger.WithError(err).Warn("Failed to decode message")
 			}
+			rpc.Messages[i].Message = &pbm{msg}
+		}
+		dump, err := json.Marshal(rpc)
+		if err != nil {
+			logger.WithError(err).Fatal("Failed to marshal rpc")
 		}
-
-		dump, _ := json.Marshal(rpc)
 		fmt.Fprintln(output, string(dump))
+		dss.Unlock()
```

Comment:
```suggestion
```

Need to delete this unlock otherwise the mutex gets unlocked twice 🙂 
e.g. https://app.circleci.com/pipelines/github/bradleyjkemp/grpc-tools/263/workflows/00349c34-d672-453c-aa22-7bd211b761f7/jobs/518/steps